contracts/auction/ClockAuction.sol

```diff
@@ -34,13 +32,15 @@ contract ClockAuction is ClockAuctionBase {
         require(candidateContract.supportsInterface(0x80ac58cd));
 
         nonFungibleContract = candidateContract;
+        registry = _registry;
 
-        _setRING(_RING);
-        _setTokenVendor(_tokenVendor);
-
+        RING = ERC20(registry.addressOf(AuctionSettingIds.CONTRACT_RING_ERC20_TOKEN));
```

Comment:
If we do a cache storage in the contract, then the RING will not get updated if the address of RING in registry get updated.